Libraries/Network/NetInfo.js

```diff
@@ -233,14 +230,25 @@ const NetInfo = {
     },
   },
 
-  isConnectionExpensive(callback: (metered: ?boolean, error?: string) => void): void {
+  isConnectionExpensive(): Promise {
+    if (arguments.length > 1) {
+      console.warn('NetInfo.isConnectionMetered(callback) is deprecated. Use the returned Promise instead.');
+      const callback = arguments[1];
+      if (Platform.OS === 'android') {
+        RCTNetInfo.isConnectionMetered().then(
+          metered => callback(metered),
+          error => callback(null, error)
+        );
+      } else {
+        callback(null, 'Unsupported');
+      }
+      return;
```

Comment:
Your have a valid argument, yeah.

My concerns:
- temporary things tend to get more permanent than we plan
- we will do this `if (callbacks.length > 0)` a lot, for every API that we want to promisify. 
If we don't provide a helper function we will have much more almost dead code in APIs for quite a while.

Let's get some more people involved in the discussion, @nicklockwood @mkonicek what do you think?